File: public/PaysPersister/lib/Classes/Poste.py
from lib.Classes.Position import Position

class Poste:
    id = None
    abreviation = None
    nom = None
    idTransferMarkt = None
    position = Position()
    
    def __init__(self):
        pass

    def toJson(self, schema:str) -> dict:
        if schema == 'persist.Joueur':
            json = {
                "id": self.id,
            }
        else :
            json = {
                "id": self.id,
                "abreviation": self.abreviation,
                "nom": self.nom,
                "id_transfermarkt": self.idTransferMarkt,
                "position": self.position.toJson(schema=schema)
            }
        
        return json

    def fromJson(self, json: dict):
        self.id = json['id'] if json.get('id') else self.id
        self.abreviation = json['abreviation'] if json.get('abreviation') else self.abreviation
        self.nom = json['nom'] if json.get('nom') else self.nom
        self.idTransferMarkt = json['id_transfermarkt'] if json.get('id_transfermarkt') else self.idTransferMarkt
        self.position = Position().fromJson(json=json['position']) if json.get('position') else self.position

        return self

File: src/Entity/Poste.php
<?php

namespace App\Entity;

use App\Repository\PosteRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

#[ORM\Entity(repositoryClass: PosteRepository::class)]
class Poste
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    #[Groups(['persist.Joueur', 'read.Joueur', "get.Poste"])]
    private $id;

    #[ORM\Column(type: 'string', length: 10, nullable: true)]
    #[Groups(["persist.Position", "get.Poste"])]
    private $abreviation;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Groups(["persist.Position", "get.Poste"])]
    private $nom;

    #[ORM\Column(type: 'integer', nullable: true)]
    #[Groups(["persist.Position", "get.Poste"])]
    private $id_transfermarkt;

    #[ORM\ManyToOne(targetEntity: Position::class, inversedBy: 'postes')]
    #[Groups(["get.Poste"])]
    private $position;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getAbreviation(): ?string
    {
        return $this->abreviation;
    }

    public function setAbreviation(?string $abreviation): self
    {
        $this->abreviation = $abreviation;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getIdTransfermarkt(): ?int
    {
        return $this->id_transfermarkt;
    }

    public function setIdTransfermarkt(?int $id_transfermarkt): self
    {
        $this->id_transfermarkt = $id_transfermarkt;

        return $this;
    }

    public function getPosition(): ?Position
    {
        return $this->position;
    }

    public function setPosition(?Position $position): self
    {
        $this->position = $position;

        return $this;
    }
}
